native/src/main/java/io/ballerina/stdlib/stan/consumer/StreamingListener.java

```diff
@@ -84,12 +85,28 @@ public void onMessage(Message msg) {
         Type[] parameterTypes = onMessageResource.getParameterTypes();
         if (parameterTypes.length == 1) {
             Object[] args1 = new Object[2];
-            args1[0] = populatedMsgRecord;
+            if (parameterTypes[0].getTag() == TypeTags.INTERSECTION_TAG) {
+                Map<String, Object> valueMap = new HashMap<>();
+                valueMap.put(Constants.MESSAGE_CONTENT, ValueCreator.createArrayValue(msg.getData()));
+                valueMap.put(Constants.MESSAGE_SUBJECT, StringUtils.fromString(msg.getSubject()));
+                args1[0] = ValueCreator.createReadonlyRecordValue(Utils.getModule(),
+                        Constants.NATS_STREAMING_MESSAGE_OBJ_NAME, valueMap);
+            } else {
+                args1[0] = populatedMsgRecord;
+            }
             args1[1] = true;
             dispatch(args1, msg.getSubject(), returnType);
         } else if (parameterTypes.length == 2) {
             Object[] args2 = new Object[4];
-            args2[0] = populatedMsgRecord;
+            if (parameterTypes[0].getTag() == TypeTags.INTERSECTION_TAG) {
+                Map<String, Object> valueMap = new HashMap<>();
+                valueMap.put(Constants.MESSAGE_CONTENT, ValueCreator.createArrayValue(msg.getData()));
+                valueMap.put(Constants.MESSAGE_SUBJECT, StringUtils.fromString(msg.getSubject()));
```

Comment:
Shall we move 
```
Map<String, Object> valueMap = new HashMap<>();
valueMap.put(Constants.MESSAGE_CONTENT, ValueCreator.createArrayValue(msg.getData()));
valueMap.put(Constants.MESSAGE_SUBJECT, StringUtils.fromString(msg.getSubject()));
```
to a separate function so that we can get rid of repeated lines?

Reply:
Fixed in c2dcb2268ce163b3d0127001ccd60c7a728872a9